core/crypto/crypto.c

```diff
@@ -8,43 +8,94 @@
 #include <crypto/aes-ccm.h>
 #include <crypto/aes-gcm.h>
 #include <crypto/crypto.h>
+#include <crypto/crypto_impl.h>
 #include <kernel/panic.h>
 #include <stdlib.h>
 #include <string.h>
 
-#if !defined(_CFG_CRYPTO_WITH_HASH)
-TEE_Result crypto_hash_alloc_ctx(void **ctx __unused, uint32_t algo __unused)
+TEE_Result crypto_hash_alloc_ctx(void **ctx, uint32_t algo)
 {
-	return TEE_ERROR_NOT_IMPLEMENTED;
+	TEE_Result res __maybe_unused = TEE_SUCCESS;
+	struct crypto_hash_ctx *c __maybe_unused = NULL;
+
+	switch (algo) {
+#if defined(CFG_CRYPTO_MD5)
```

Comment:
Minor - how about moving the `#ifdef`s to `crypto_impl.h`? (the less conditionals in C files, the better).

Reply:
I was considering the same, I'll fix.